ibis/backends/base/__init__.py

```diff
@@ -19,16 +19,68 @@ def name(self) -> str:
         """
         pass
 
+    @property
+    @abc.abstractmethod
```

Comment:
shouldn't this be an abstractproperty?

Reply:
`abstractproperty` was deprecated since version 3.3 in favor of this. Apparently it was implemented because `property` didn't work with `abstractmethod`. See https://docs.python.org/3/library/abc.html#abc.abstractproperty